docs/sidecar/security.md

```diff
@@ -0,0 +1,628 @@
+- Encrypt Redis connection (use SSL/TLS)
+- Use Redis authentication
+- Configure Redis network policies
+- Set appropriate token cache expiration
+- Monitor cache access patterns
+
+### Token Validation
+
+Configure strict token validation:
+
+```yaml
+- name: AzureAd__Audience
+  value: "api://<your-api-id>"  # Validate expected audience
+
+- name: AzureAd__Scopes
+  value: "access_as_user"  # Enforce required scopes
+```
+
+## Container Security
+
+### Run as Non-Root User
+
+Configure the sidecar to run as a non-root user:
+
+```yaml
+containers:
+- name: sidecar
+  image: mcr.microsoft.com/identity/sidecar:latest
+  securityContext:
+    runAsNonRoot: true
+    runAsUser: 1000
+    runAsGroup: 3000
+    allowPrivilegeEscalation: false
+    capabilities:
+      drop:
+      - ALL
+    readOnlyRootFilesystem: true
+    seccompProfile:
+      type: RuntimeDefault
+```
+
+### Pod Security Standards
+
+Apply Pod Security Standards:
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: myapp-namespace
+  labels:
+    pod-security.kubernetes.io/enforce: restricted
+    pod-security.kubernetes.io/audit: restricted
+    pod-security.kubernetes.io/warn: restricted
+```
+
+### Image Security
+
+1. **Use Official Images**: Only use images from `mcr.microsoft.com/identity/sidecar`
+2. **Pin Versions**: Use specific version tags, not `latest`
+3. **Scan Images**: Regularly scan for vulnerabilities
+4. **Verify Signatures**: Validate image signatures before deployment
+
+```yaml
+containers:
+- name: sidecar
+  image: mcr.microsoft.com/identity/sidecar:1.0.0  # Specific version
+  imagePullPolicy: Always
+```
+
+## Monitoring and Auditing
+
+### Enable Comprehensive Logging
+
+```yaml
+- name: Logging__LogLevel__Default
+  value: "Information"
+- name: Logging__LogLevel__Microsoft.Identity.Web
+  value: "Information"
+
+# Avoid Debug in production (may log sensitive data)
+# - name: Logging__LogLevel__Microsoft.Identity.Web
+#   value: "Debug"  # ❌ Not for production
+```
+
+### Application Insights Integration
+
+```yaml
+- name: ApplicationInsights__ConnectionString
+  valueFrom:
+    secretKeyRef:
+      name: monitoring-secrets
+      key: appinsights-connection
+```
+
+### Audit Token Acquisition
+
+Log and monitor:
+- Token acquisition requests
+- Failed authentication attempts
+- Agent identity usage
+- Unusual access patterns
+- Token acquisition latency
+
+### Security Alerts
+
+Configure alerts for:
+- Repeated authentication failures
+- Token acquisition errors
+- Unusual API access patterns
+- Network policy violations
+- Certificate expiration warnings
+
+## Resource Limits
+
+Set appropriate resource limits to prevent resource exhaustion attacks:
+
+```yaml
+containers:
+- name: sidecar
+  resources:
+    requests:
+      memory: "128Mi"
+      cpu: "100m"
+    limits:
+      memory: "256Mi"
+      cpu: "250m"
+```
+
+## Agent Identity Security
+
+### Validate Agent Parameters
+
+Implement validation for agent identity parameters:
```

Comment:
Perhaps more important to mention, don't use unvalidated user input for any of the options to the container API.